src/components/ProtectedImage.vue

```diff
@@ -0,0 +1,26 @@
+<script lang="ts" setup>
+import filters from "@/utils/filters";
+import { useAuth0 } from "@auth0/auth0-vue";
+import { onMounted, ref } from "vue";
+
+const props = defineProps<{
+  src: string | null | undefined;
```

Comment:
Would it be better not to allow `null` or `undefined` here? There won't be an image to show in that case anyways and the template including this one would have to take care about this (either not showing the image or using a placeholder or whatever fits the case).

Reply:
I guess implementing the placeholder outside is better. Originally thought of doing that internally in the component.